tests/lcp/test_collection.py

```diff
@@ -0,0 +1,516 @@
+        lcp_server_mock.generate_license = MagicMock(return_value=lcp_license)
+
+        with self._configuration_factory.create(
+                self._configuration_storage, self._db, LCPServerConfiguration) as configuration:
+            with patch('api.lcp.collection.LCPServer') as lcp_server_constructor:
+                lcp_server_constructor.return_value = lcp_server_mock
+
+                configuration.lcpserver_url = fixtures.LCPSERVER_URL
+                configuration.lcpserver_user = fixtures.LCPSERVER_USER
+                configuration.lcpserver_password = fixtures.LCPSERVER_PASSWORD
+                configuration.lcpserver_input_directory = fixtures.LCPSERVER_INPUT_DIRECTORY
+
+                configuration.provider_name = fixtures.PROVIDER_NAME
+                configuration.passphrase_hint = fixtures.TEXT_HINT
+                configuration.encryption_algorithm = LCPServerConfiguration.DEFAULT_ENCRYPTION_ALGORITHM
+                configuration.max_printable_pages = LCPServerConfiguration.DEFAULT_MAX_PRINTABLE_PAGES
+                configuration.max_copiable_pages = LCPServerConfiguration.DEFAULT_MAX_COPIABLE_PAGES
+
+                # Act
+                license_pool.loan_to(patron, start=today, end=expires, external_identifier=lcp_license['id'])
+                fulfilment_info = lcp_api.fulfill(patron, 'pin', license_pool, 'internal format')
+
+                # Assert
+                eq_(fulfilment_info.collection_id, self._lcp_collection.id)
+                eq_(fulfilment_info.collection(self._db), self._lcp_collection)
+                eq_(fulfilment_info.license_pool(self._db), license_pool)
+                eq_(fulfilment_info.data_source_name, data_source_name)
+                eq_(fulfilment_info.identifier_type, license_pool.identifier.type)
+
+    def test_patron_activity_returns_correct_result(self):
+        # Arrange
+        lcp_api = LCPAPI(self._db, self._lcp_collection)
+
+        patron = self._patron()
+        days = self._lcp_collection.default_loan_period(patron.library)
+        today = datetime.datetime.utcnow()
+        expires = today + datetime.timedelta(days=days)
+        data_source = DataSource.lookup(self._db, DataSource.LCP, autocreate=True)
+        data_source_name = data_source.name
+        external_identifier = '1'
+        license_pool = self._licensepool(
+            edition=None, data_source_name=data_source_name, collection=self._lcp_collection)
+        license_pool.loan_to(patron, start=today, end=expires, external_identifier=external_identifier)
+
+        other_collection = self._collection(protocol=ExternalIntegration.MANUAL)
+        other_external_identifier = '2'
+        other_license_pool = self._licensepool(
+            edition=None, data_source_name=data_source_name, collection=other_collection)
+        other_license_pool.loan_to(patron, start=today, end=expires, external_identifier=other_external_identifier)
```

Comment:
This test should also create other loans that ought not to show up in the list:

* A loan that's out to some other patron than `patron`.
* A loan that has expired (assuming you don't want to include those -- see https://github.com/NYPL-Simplified/circulation/pull/1472/files#r476572493)
* Optionally, a loan that hasn't started yet (assuming you don't want to include those).

Reply:
Added additional test cases